buildkite/resource_test_suite_test.go

```diff
@@ -0,0 +1,175 @@
+package buildkite
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAcc_testSuiteAddRemove(t *testing.T) {
+	t.Parallel()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: protoV6ProviderFactories(),
+		CheckDestroy:             testTestSuiteDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: `
+				resource "buildkite_team" "team" {
+					name = "test suite team"
+					default_team = false
+					privacy = "VISIBLE"
+					default_member_role = "MAINTAINER"
+				}
+				resource "buildkite_test_suite" "suite" {
+					name = "test suite"
+					default_branch = "main"
+					team_owner_id = resource.buildkite_team.team.id
+				}
+				`,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet("buildkite_test_suite.suite", "id"),
+					resource.TestCheckResourceAttrSet("buildkite_test_suite.suite", "api_token"),
+					resource.TestCheckResourceAttr("buildkite_test_suite.suite", "default_branch", "main"),
+					resource.TestCheckResourceAttr("buildkite_test_suite.suite", "name", "test suite"),
+					resource.TestCheckResourceAttrSet("buildkite_test_suite.suite", "team_owner_id"),
+					checkTestSuiteExists("buildkite_test_suite.suite"),
+					checkTestSuiteRemoteValue("buildkite_test_suite.suite", "Name", "test suite"),
+					checkTestSuiteRemoteValue("buildkite_test_suite.suite", "DefaultBranch", "main"),
```

Comment:
I've implemented remote testing in this way to see how it feels. Instead of the first function here setting a pointer variable and remaining functions using that pointer to check values, it retrieves the suite each time. It's quite slow and I'm not sure I like it much. Let me know what you think

Reply:
I don't mind it, but if one of our pieces of upcoming work is speeding up the tests and properly implementing `Parallel()` then opting for a testing approach which is slower might go against this.

I'm indifferent to whether it stays in or changes to a pointer approach, but it feels like that's what's going to happen anyway and it might be good to have that implemented so that core tests don't need to be changed, merely out use of `Parallel`.

Reply:
Yeah I don't like it very much. I think there is some improvement we could do to the testing structure like this but not quite this. I think it would still involve pointers.
I'll convert this back to the other approach